torch/csrc/jit/ir.h

```diff
@@ -1294,5 +1315,25 @@ TORCH_API std::vector<Value*> inlineCallTo(
     ArrayRef<Value*> inputs,
     bool unpack_outputs = false);
 
+// this holds profiling information for a single execution of a graph
+// Note, since it's using ProfiledTensorTypePtr we could re-use
+// the same ProfilingRecord for multiple runs ProfiledTensorTypePtr takes
+// care of merging the values
+struct ProfilingRecord {
+
+
+  ProfilingRecord(const std::shared_ptr<Graph>& g): profiled_graph_(g) {}
+  ProfilingRecord(const ProfilingRecord& other);
+
+  std::unique_ptr<ProfilingRecord> copy () const;
+  std::unordered_map<size_t, ProfiledTensorTypePtr> profiled_types_;
+  std::shared_ptr<Graph> profiled_graph_;
+  void record (const IValue& ival, size_t unique);
```

Comment:
I think you'll want to record Value -> IValue?

Reply:
`unique -> value` is more flexible (described in the other comment). A consumer of profiling information will still pass `Value*` from which we extract its unique id and locate the corresponding profiling info.